### ISSUE TITLE
bug: datetime month/year picker is hard to use with screen readers

### ISSUE THREAD

**Issue by amandaejohnston** [ionic-team/ionic-framework#25221]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When using `ion-datetime`'s month/year picker with a screen reader, swiping to get around the page, it isn't possible to close the picker without unintentionally changing your selection. You have to swipe past all months to get back to the picker, and the scrolling automatically changes which month is selected.

Screencast: https://user-images.githubusercontent.com/90629384/166057993-70e1c6ad-2053-431c-a908-4ef03787fac7.mov

### Expected Behavior

You should be able to change the month/year on a screen reader to any value. I'm not sure what should change about the design, though. Pressing Escape to toggle the picker would work on desktop, but not on phones, for example.

### Steps to Reproduce

Use any `ion-datetime` with a presentation that shows the month/year picker as a separate overlay.

### Ionic Info

Recording was from core hosted through ngrok; output from hosting PC:
```
[WARN] You are not in an Ionic project directory. Project context may be missing.

Ionic:

   Ionic CLI : 6.18.1

Utility:

   cordova-res : not installed globally
   native-run  : 1.4.1

System:

   NodeJS : v16.13.0
   npm    : 8.1.0
   OS     : Windows 10
```

**Comment by liamdebeasi** [comment]
Native iOS does something similar that we can probably follow:

https://user-images.githubusercontent.com/2721089/166465753-4fa4a0c7-74f6-4067-8ae7-d2c236c9330d.mov

With VoiceOver enabled, users can select the column as a whole. In order to interact with the picker, they need to double tap the column. At this point, users are able to drag the column up or down.

**Comment by iamandersonp** [comment]
This same behavior ocurs for the ion-datetime in wheel mode

in the video of [liamdebeasi](https://github.com/liamdebeasi) you can hear the voiceover say the month and year are sliders but the actual month selector is readed as buttons and nothin hapens when you select those buttons

**Comment by liamdebeasi** [comment]
This will be fixed in an upcoming minor release of Ionic 8: https://github.com/ionic-team/ionic-framework/pull/29371
